haskell/private/actions/compile.bzl

```diff
@@ -210,7 +210,7 @@ def _compilation_defaults(hs, cc, java, dep_info, srcs, extra_srcs, cpp_defines,
     DefaultCompileInfo: Populated default compilation settings.
   """
   args = hs.actions.args()
-  haddock_args = hs.actions.args()
+  ghc_args = []
```

Comment:
Here's a refactoring suggestion; up to you whether it's worth it: consider moving l.272-282 below.  Nearly all of args and ghc_args seem duplicated; it seems like we should be able to structure the code like:

```
ghc_args = []
ghc_args += ["-hide-all-packages"]
ghc_args += ...
...

args = hs.actions.args()
args.add(ghc_args)
# then add -hidir, -odir, etc to args
```

In theory it might even be possible to extract the computation of ghc_args into a separate function.  Again, up to you whether it's worth it at this point.